### PR TITLE
handle named space packages

### PR DESCRIPTION
**Pull request by ocefpaf**
~~No ready for review yet. (Missing step 3 from https://github.com/ericdill/depfinder/issues/22#issuecomment-431368167)~~

Ready for review!

### CODE REVIEW

**Comment by ericdill** [comment]
> @ericdill and @tonyfast for step 3 what do you have in mind? Parse from a simple text file? Or maybe use yaml or json? Single file with all 3 categories (name space, package mapping and fake packages) or 3 different files?

I think starting simple is probably the best, so one file is my vote. I'd say JSON if we didn't already have a pyyaml dependency. Up to you.

**Comment by ocefpaf** [comment]
Single file and yaml it is :wink:

(I'll try to finish this tomorrow.)

File: depfinder/pkg_data/pkg_data.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  sklearn: scikit-learn
+  stdlib_list: stdlib-list
+  yaml: pyyaml
+
+_NAMEDSPACE_MAPPING:
+  ruamel.yaml: ruamel.yaml
+
+
+_FAKE_PACKAGES:
+  matplotlib: !!set {mpl_toolkits: null}
```
Comment (ocefpaf) [inline depfinder/pkg_data/pkg_data.yml]: I don't really like the `!!set {value: null}` syntax but I'm not a `yaml` expert to figure out a better way. Maybe we can change the logic in `depfinder` to take lists there if that makes sense. (But I would leave that for another PR :smile:)

Comment (ericdill) [reply depfinder/pkg_data/pkg_data.yml]: I agree that this is kind of gross syntax. I also agree that changing this to lists can wait for another PR.

File: depfinder/main.py [hunk 2]
```diff
@@ -59,6 +59,18 @@
 AST_QUESTIONABLE = tuple(list(AST_TRY) + [ast.FunctionDef, ast.ClassDef])
 del AST_TRY
 
+PACKAGE_NAME = None
+
+_HERE = os.path.dirname(os.path.abspath(__file__))
+with open(os.path.join(_HERE, 'pkg_data', 'pkg_data.yml')) as f:
+    pkg_data = yaml.load(f)
```
Comment (ericdill) [inline depfinder/main.py]: Did you consider using `pkgutil`? 

e.g.:
```python
In [2]: import yaml

In [3]: import pkgutil

In [4]: yaml.load(pkgutil.get_data("depfinder", "pkg_data/pkg_data.yml").decode())
Out[4]:
{'_PACKAGE_MAPPING': {'av': 'pyav',
  'bs4': 'beautifulsoup4',
  'cv2': 'opencv',
  'IPython': 'ipython',
  'netCDF4': 'netcdf4',
  'PIL': 'pillow',
  'skimage': 'scikit-image',
  'sklearn': 'scikit-learn',
  'stdlib_list': 'stdlib-list',
  'yaml': 'pyyaml'},
 '_NAMEDSPACE_MAPPING': {'ruamel.yaml': 'ruamel.yaml'},
 '_FAKE_PACKAGES': {'matplotlib': {'mpl_toolkits'},
  'pymongo': {'bson', 'gridfs'}}}
```

**Comment by ericdill** [comment]
LGTM. Thanks @ocefpaf . We can address any comments in follow-on PRs

File: depfinder/main.py (hunk 2 above)
Comment (ocefpaf) [reply depfinder/main.py]: Nope. I usually copy-n-paste that pattern and I never tried it. It is definetly better. I'll send another PR soon. Thanks!

Comment (ericdill) [reply depfinder/main.py]: I suspect that

```python
pkgutil.get_data("depfinder", "pkg_data/pkg_data.yml"
```

should become

```python
pkgutil.get_data(__name__, "pkg_data/pkg_data.yml"
```

if the code is meant to be used in the same package.

> I'll send another PR soon

Sounds good. But no rush. Or I'll do it at some point :)  Thanks for all the contributions so far!
